index.js

```diff
@@ -319,33 +324,59 @@ module.exports = {
     }
 
     function listMembers(groupId, opts = {}) {
-      return pull(
-        getPickedEpoch(groupId, { live: !!opts?.live }),
-        pull.map((pickedEpoch) => {
-          console.log('pickedEPoch', pickedEpoch)
-          return getMemberUpdates(pickedEpoch.id, { live: !!opts?.live })
-        }),
-        pull.flatten(),
-        pull.map((update) => {
-          if (update.added) return update.added
-          return update
-        }),
-        // if it's not live, only include adds
-        pull.filter((update) => opts?.live || !update.excluded)
-      )
+      const { live } = opts
+      const deferredSource = pullDefer.source()
+
+      get(groupId, (err, group) => {
+        // prettier-ignore
+        if (err) return deferredSource.abort(clarify(err, 'Failed to get group info when listing members'))
+        // prettier-ignore
+        if (group.excluded) return deferredSource.abort( new Error("We're excluded from this group, can't list members"))
+
+        if (!live) {
+          getPreferredEpoch(groupId, (err, epoch) => {
+            // prettier-ignore
+            if (err) return deferredSource.abort(clarify(err, 'failed to load preferred epoch'))
+
+            getMembers(epoch.id, (err, res) => {
+              // prettier-ignore
+              if (err) return deferredSource.abort(clarify(err, 'error getting members'))
+
+              const source = pull.once(res)
+              deferredSource.resolve(source)
+            })
+          })
+          return
+        }
+
+        let abortable = pullAbortable()
+        const source = pull(
+          getPreferredEpoch.stream(groupId, { live }),
+          pull.map((epoch) => {
+            abortable.abort()
```

Comment:
added killing the old stream after we finds a new epoch

Reply:
woa this was confusing at first but ok you sort of reuse the same aborter. and you don't want getMembers to necessarily end automatically so you do the aborting here